sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -229,16 +228,44 @@ public virtual Response Upload<T>(string ruleId, string streamName, IEnumerable<
 
                     if (response.Status != 204)
                     {
-                        throw new RequestFailedException(response);
+                        // if there is no Handler on options, throw exception otherwise raise Handler
+                        if (options == null || !options.HasHandler)
```

Comment:
Can we use the same options approach as in the async method, i.e.
```c#
options ??= new Options();
```

Reply:
we could, we would just not use the Concurrency param. I'll add it so it's the same in both cases

Reply:
It doesn't look like this was updated.

Reply:
missed two! sorry

Reply:
I don't see any null coalescing statement though.

Reply:
Ah, yes. It's coming back to me. Pavel initially defaulted this to `false` because so many would've failed, but once we got clean we - the client library owners - need to set it `true` e.g., https://github.com/Azure/azure-sdk-for-net/pull/27551/files#diff-d6d7fdd20f8d32d2438f28c46f37f8bb028a8873ddfa90a12eb6bad3765159bbR29. IIRC, there was an issue on getting clean and enabling. I'll see if I can find and resurrect it; otherwise, I'll create a new one. We should drive to make this `true` by default.